### PR TITLE
feat: Backlog/ci lib to pypi on tag

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves : 

* CLICKUP-
* FT-d9b07b65-2f02-4140-bacd-eb76e20b8313
* SENTRY-
* ZENDESK-

- [ ] I have added automatic tests where applicable.
- [ ] The PR contains a description of what has been changed.
- [ ] The description contains manual test instructions.
- [ ] The PR contains update to the release notes.
- [ ] The PR contains update to the documentation.

This PR has been tested on :

- [ ] Windows.
- [ ] MacOs.
- [ ] Linux.

### CODE REVIEW

File: .github/workflows/push-tag-release.yml
Comment (jimmycallin) [inline .github/workflows/push-tag-release.yml:9]: is there anyway to filter this more? for instance, we always use `v*` in other projects

File: .github/workflows/push-tag-release.yml
Comment (jimmycallin) [inline .github/workflows/push-tag-release.yml:42]: remove comments

File: tools/build.py
Comment (jimmycallin) [inline tools/build.py:587]: remove comments

File: .github/workflows/push-tag-release.yml
Comment (jimmycallin) [inline .github/workflows/push-tag-release.yml:42]: remove comments

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:222]: artefacts is more british/commonwealth:ish, while artifacts is US spelling. What is our code standard preference here?

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:200]: Fair point, we are actually moving over to Python 3.9

**Review by lluisFtrack — DISMISSED** [review]
Can't test it.

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [inline .github/workflows/push-tag-release.yml:77]: I think we should by default always do the following steps (Which if I understood it correctly is almost what you already do):
- checkout
- setup poetry
- poetry_build
- build_connect_package
* if there is any special package that has to follow another setup like rv, we add an if for the rv and we build the rv-plugin.

Then theoretically we shouldn't be in need of building the style when want to release PS right? It should only be a requirement when releasing the qt-style library, And that could be an if somewhere. 
But in any case if that is needed when releasing any framework package or we have to release a different package depending on the OS, then I think we should do it with a matrix like we did in the old release.yml file where we specify the OS and tag the package with a suffix, so we don't have to duplicate the code.

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [inline .github/workflows/push-tag-release.yml:103]: Same comment as above

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [inline .github/workflows/push-tag-release.yml:162]: Shouldn't this be build only when releasing qt-style? I might be wrong, but then if we have to build it every time we release PS, we should download it from the published library qt-style right?

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [inline .github/workflows/push-tag-release.yml:254]: I can see in the backend example they have to checkout and setup python, isn't that needed?
Soruce: https://github.com/ftrackhq/ftrack-python/blob/main/.github/workflows/cicd.yml
```
- uses: actions/checkout@v1
      - name: Set up Python
        uses: actions/setup-python@v4
        with:
          python-version: "3.x"
```

**Review by lluisFtrack — COMMENTED** [review]
If I'm not wrong, we agreed to follow the same workflow as in backend, are you sure. that we are doing this? ( https://backlightco.slack.com/archives/C05G1UAE1LY/p1704808677270209 )

- You create a release in github (which creates a tag)
- It publishes to test then holds waiting for approval
- If all looks ok then you trigger the publish to prod by approving the deployment

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:46]: Let's discuss this with the team, it is a good solution to the problem but we step away from the official monorepo tag name convention. After some reading online, one alternative solution is to use path/to/package@vX.Y.Z.

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:9]: Will re-create the CI to trig on release creation, then it is up to the user to drive this.

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:65]: Will be removed

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:77]: The CI is designed to build all plugins on Linux that are platform independent, for example Maya in the future.

This if gate is simply because Photoshop is not supported on Linux, and is a special case. IMO it is fine for now to have these if gates, instead of complicating the CI with matrixes that are hard to interpret and also have limitations - can recall you cannot drive the OS it runs on through matrix variables.

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:162]: It is only built if it is the qt-style yes, notice the ==

File: .github/workflows/push-tag-release.yml
Comment (henriknorin-ftrack) [reply .github/workflows/push-tag-release.yml:254]: We don't need this as we already have built the artefact, we just pull it from temp artefact storage

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [reply .github/workflows/push-tag-release.yml]: Ok, But yes, you can drive the OS in the matrix as well, you have an example in the removed release.yaml file:

```
    strategy:
      matrix:
        # Add new packages here
        package: [application-launcher, connect-action-launcher-widget, connect-plugin-manager, framework-core, framework-qt, framework-maya, framework-nuke, framework-houdini, framework-3dsmax, connect-timetracker-widget, nuke-studio, rv]
        os: [ubuntu-latest]
        suffix: ['']
        include:
          - package: framework-unreal
            os: windows-latest
            suffix: 'win'
          # Add platform dependent packages here
```

File: .github/workflows/push-tag-release.yml
Comment (lluisFtrack) [reply .github/workflows/push-tag-release.yml]: It has to be removed form here as it has the if PS package before

**Review by lluisFtrack — COMMENTED** [review]
I think is worth re-review this on a later stage to try to improve it and simplify it with Matrix.

**Pull request merged**
